conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandlerTest.java

```diff
@@ -80,6 +86,74 @@ public void handlesServiceException() throws IOException {
                 .contains("\"parameters\":{\"foo\":\"bar\"}}");
     }
 
+    @Test
+    public void checkServiceExceptionArgumentsSerializedWithToString() throws IOException {
+        record Argument(
+                String field1, Optional<Integer> maybeField2, List<String> collectionField, String nullString) {}
+        exception = new ServiceException(
+                ErrorType.CONFLICT,
+                SafeArg.of("arg", new Argument("foo", Optional.of(42), List.of("bar", "baz"), null)));
+        HttpURLConnection connection = execute();
+
+        assertThat(connection.getResponseCode()).isEqualTo(ErrorType.CONFLICT.httpErrorCode());
+        String expectedSerializedArg =
+                "\"Argument[field1=foo, maybeField2=Optional[42], collectionField=[bar, baz], nullString=null]\"";
+        assertThat(getErrorBody(connection))
+                .contains("{\"errorCode\":\"CONFLICT\"")
```

Comment:
We should verify the errorName in addition to the errorCode (name is arguably much more important!)
parsing to either `ConjureError` or a locally-defined variant for testing will make that more robust than string matching json contents

Reply:
Updated to also check the error name.